lib/spack/spack/solver/counter.py

```diff
@@ -34,6 +34,9 @@ def __init__(self, specs: List["spack.spec.Spec"], tests: bool) -> None:
 
         self._possible_dependencies: PossibleDependencies = set()
         self._possible_virtuals: Set[str] = set(x.name for x in specs if x.virtual)
+        for x in specs:
+            if not x.virtual:
+                self._possible_virtuals.update(str(i) for i in x.package_class.provided.keys())
```

Comment:
Ok, this would be part of the changes in semantic mentioned above. Not sure why we would need it though :thinking: